apps/filebrowser/src/filebrowser/api.py

```diff
@@ -43,12 +44,22 @@ def decorator(*args, **kwargs):
 @error_handler
 def get_filesystems(request):
   response = {}
+  filesystems = []
+  user_home_dir = ''
 
-  filesystems = {}
-  for k in fsmanager.get_filesystems(request.user):
-    filesystems[k] = True
+  for fs in fsmanager.get_filesystems(request.user):
+    if fs == 'hdfs':
+      user_home_dir = request.user.get_home_directory()
+    elif fs == 's3a':
+      user_home_dir = get_s3_home_directory(request.user)
+    elif fs == 'abfs':
+      user_home_dir = get_home_dir_for_abfs(request.user)
 
-  response['status'] = 0
-  response['filesystems'] = filesystems
+    filesystems.append({
+      fs: {
+        'user_home_directory': user_home_dir,
+      }
+    })
 
+  response['filesystems'] = filesystems
```

Comment:
No need for the filesystems attribute, i.e. return JsonResponse(filesystems)

Reply:
Done